### PR TITLE
Updated to ginkgo v2 in mcm

### LINKED ISSUE #857: Upgrade to ginkgo v2.x
**How to categorize this issue?**

/area dev-productivity
/kind enhancement
/priority 2

**What would you like to be added**:

Upgrade the IT of mcm-providers + framework part of MCM to use ginkgo v2.x
Pick up only after #853 as its higher prio + to avoid any pipeline IT problems

- [x] mcm (https://github.com/gardener/machine-controller-manager/pull/870)
- [x] provider-aws(https://github.com/gardener/machine-controller-manager-provider-aws/pull/151)
- [x] provider-az(https://github.com/gardener/machine-controller-manager-provider-azure/pull/127)
- [x] provider-gcp(https://github.com/gardener/machine-controller-manager-provider-gcp/pull/106)
- [x] provider-alicloud(https://github.com/gardener/machine-controller-manager-provider-alicloud/pull/69)
- [x] provider-openstack(https://github.com/gardener/machine-controller-manager-provider-openstack/pull/112)

**Why is this needed**:
To adapt to latest ginkgo changes. Since other projects have already migrated, its becoming painful to work on both simultaneously due to need to keep on changing the version.

### PR DESCRIPTION
**Pull request by piyuagr**
**What this PR does / why we need it**:
This PR upgrades the Ginkgo testing framework to v2 

**Which issue(s) this PR fixes**:
Fixes part of #857 

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,8 +8,8 @@ require (
 	github.com/davecgh/go-spew v1.1.1
 	github.com/go-git/go-git/v5 v5.4.2
 	github.com/google/uuid v1.3.0
-	github.com/onsi/ginkgo v1.16.2
-	github.com/onsi/gomega v1.27.6
+	github.com/onsi/ginkgo/v2 v2.13.1 // indirect
```
Comment (himanshu-kun) [inline go.mod:11]: isn't ginkgo v2 a direct dependency?

**Review by himanshu-kun — APPROVED** [review]
/lgtm

File: go.mod [hunk 2]
```diff
@@ -27,6 +27,8 @@ require (
 	sigs.k8s.io/yaml v1.3.0
 )
 
+require github.com/onsi/ginkgo v1.16.5
```
Comment (rishabh-11) [inline go.mod:30]: Why is Ginkgo v1 still in the go.mod? Can you remove it?

**Review by rishabh-11 — CHANGES_REQUESTED** [review]
Thanks for the PR. Please address the review comments

File: LICENSE.md [hunk 3]
```diff
@@ -286,12 +286,12 @@ https://github.com/modern-go/reflect2.
...
 Ginkgo.  
-https://github.com/onsi/ginkgo.  
+https://github.com/onsi/ginkgo/v2.  
 Copyright (c) 2013-2014 Onsi Fakhouri
-MIT license (https://github.com/onsi/ginkgo/blob/master/LICENSE)  
+MIT license (https://github.com/onsi/ginkgo/v2/blob/master/LICENSE)  
 
 Gomega.  
-https://github.com/onsi/ginkgo.  
+https://github.com/onsi/ginkgo/v2.
```
Comment (rishabh-11) [inline LICENSE.md:294]:
```suggestion
https://github.com/onsi/ginkgo.  
```

**Review by rishabh-11 — APPROVED** [review]
/lgtm

**Pull request merged**